test/Conversion/CalyxToFSM/materialize-fsm.mlir

```diff
@@ -0,0 +1,121 @@
+// RUN: circt-opt -pass-pipeline='calyx.program(calyx.component(materialize-calyx-to-fsm))' %s | FileCheck %s
+
+// CHECK: fsm.machine @control(%[[A_DONE:.*]]: i1, %[[B_DONE:.*]]: i1, %[[COND_DONE:.*]]: i1, %[[TOP_LEVEL_GO:.*]]: i1) -> (i1, i1, i1, i1) attributes {compiledGroups = [@A, @B, @cond], initialState = "[[FSM_ENTRY:.*]]"} {
+// CHECK-NEXT:   %[[C1:.*]] = hw.constant true
+// CHECK-NEXT:   %[[C0:.*]] = hw.constant false
+// CHECK-NEXT:   fsm.state @[[FSM_ENTRY]] output {
+// CHECK-NEXT:     fsm.output %[[C0]], %[[C0]], %[[C0]], %[[C0]] : i1, i1, i1, i1
+// CHECK-NEXT:   } transitions {
+// CHECK-NEXT:     fsm.transition @[[SEQ_0_COND:.*]] guard {
+// CHECK-NEXT:       fsm.return %[[TOP_LEVEL_GO]]
+// CHECK-NEXT:     }
+// CHECK-NEXT:   }
+// CHECK-NEXT:   fsm.state @[[FSM_EXIT:.*]] output {
+// CHECK-NEXT:     fsm.output %[[C0]], %[[C0]], %[[C0]], %[[C1]] : i1, i1, i1, i1
+// CHECK-NEXT:   }
+// CHECK-NEXT:   fsm.state @[[SEQ_1_IF:.*]] output {
+// CHECK-NEXT:     fsm.output %[[C0]], %[[C0]], %[[C0]], %[[C0]] : i1, i1, i1, i1
+// CHECK-NEXT:   } transitions {
+// CHECK-NEXT:     fsm.transition @[[SEQ_1_IF_THEN_A:.*]] guard {
+// CHECK-NEXT:       fsm.return %lt_reg.out
+// CHECK-NEXT:     }
+// CHECK-NEXT:     fsm.transition @[[SEQ_1_IF_ELSE_B:.*]] guard {
+// CHECK-NEXT:       %true_2 = hw.constant true
+// CHECK-NEXT:       %0 = comb.xor %lt_reg.out, %true_2 : i1
+// CHECK-NEXT:       fsm.return %0
+// CHECK-NEXT:     }
+// CHECK-NEXT:   }
+// CHECK-NEXT:   fsm.state @[[SEQ_1_IF_THEN_A]] output {
+// CHECK-NEXT:     fsm.output %[[C1]], %[[C0]], %[[C0]], %[[C0]] : i1, i1, i1, i1
+// CHECK-NEXT:   } transitions {
+// CHECK-NEXT:     fsm.transition @[[FSM_EXIT]] guard {
+// CHECK-NEXT:       fsm.return %[[A_DONE]]
+// CHECK-NEXT:     }
+// CHECK-NEXT:   }
+// CHECK-NEXT:   fsm.state @[[SEQ_1_IF_ELSE_B]] output {
+// CHECK-NEXT:     fsm.output %[[C0]], %[[C1]], %[[C0]], %[[C0]] : i1, i1, i1, i1
+// CHECK-NEXT:   } transitions {
+// CHECK-NEXT:     fsm.transition @[[FSM_EXIT]] guard {
+// CHECK-NEXT:       fsm.return %[[B_DONE]]
+// CHECK-NEXT:     }
+// CHECK-NEXT:   }
+// CHECK-NEXT:   fsm.state @[[SEQ_0_COND]] output {
+// CHECK-NEXT:     fsm.output %[[C0]], %[[C0]], %[[C1]], %[[C0]] : i1, i1, i1, i1
+// CHECK-NEXT:   } transitions {
+// CHECK-NEXT:     fsm.transition @[[SEQ_1_IF]] guard {
+// CHECK-NEXT:       fsm.return %[[COND_DONE]]
+// CHECK-NEXT:     }
+// CHECK-NEXT:   }
+// CHECK-NEXT: }
+
+calyx.program "main" {
```

Comment:
Aren't we getting rid of `calyx.program` in favor of `builtin.module`? It might be better to write the test cases without calyx.program, and just have calyx.component at the top-level of the file (in an implicit builtin.module).

Reply:
Yup - this commit is already outdated due to 6838ec2173008ca03d1678ca0eb8681b323a7892 now being merged. Tests will be updated before a merge.